src/app/shared/services/dynamic-data/adapters/persistedMemory.ts

```diff
@@ -140,7 +140,22 @@ export class PersistedMemoryAdapter {
     this.persistStateToDB();
   }
 
-  public delete(flow_type: FlowTypes.FlowType, flow_name: string, ids?: string[]) {
+  /** Set data for a given entry. Will overwrite any existing data */
+  public set(update: IDataUpdate) {
+    const { flow_name, flow_type, id, data } = update;
+    if (!this.state[flow_type]) this.state[flow_type] = {};
+    if (!this.state[flow_type][flow_name]) this.state[flow_type][flow_name] = {};
+    // Remove any values marked as undefined
+    for (const [key, value] of Object.entries(data)) {
+      if (value === undefined) {
+        delete data[key];
+      }
+    }
+    this.state[flow_type][flow_name][id] = data;
+    this.persistStateToDB();
```

Comment:
Should this be async/awaited so the `set()` method can also be awaited?

Reply:
not on this occasion, as it's part of a synchronous `set` method which is designed to update the in-memory value, and let the persistence just happen as a side-effect (trigger but don't wait to complete). I can't actually remember why I wanted the sync set/update methods, but better to keep non-async as persistence always has a bit of debounce/delay that isn't super important to whatever prompted the update